c8y panics when session variables are not set

The following panics due to accessing non existent URL struct when creating the c8yclient.

```sh
Installing bash completion for aliases
/root/.cumulocity/c8y-admin.json
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x605500]

 

goroutine 1 [running]:
net/url.(*URL).String(0x0)
        net/url/url.go:810 +0x40
github.com/reubenmiller/go-c8y-cli/v2/pkg/cmd/factory.CreateCumulocityClient.func1()
        github.com/reubenmiller/go-c8y-cli/v2/pkg/cmd/factory/c8yclient.go:183 +0xc2b
github.com/reubenmiller/go-c8y-cli/v2/pkg/cmd/sessions/set.(*CmdSet).RunE(0xc0001f33b0, 0x0?, {0xc0008210a0, 0x0, 0x1})
        github.com/reubenmiller/go-c8y-cli/v2/pkg/cmd/sessions/set/set.manual.go:149 +0x356
github.com/reubenmiller/go-c8y-cli/v2/pkg/cmd/subcommand.NewSubCommand.func1(0xc000386c00?, {0xc0008210a0, 0x0, 0x1})
        github.com/reubenmiller/go-c8y-cli/v2/pkg/cmd/subcommand/subcommand.go:55 +0x6d
github.com/spf13/cobra.(*Command).execute(0xc000386c00, {0xc000821080, 0x1, 0x1})
        github.com/spf13/cobra@v1.7.0/command.go:940 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0xc000004300)
        github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.7.0/command.go:992
github.com/reubenmiller/go-c8y-cli/v2/pkg/cmd.MainRun()
        github.com/reubenmiller/go-c8y-cli/v2/pkg/cmd/cmd.go:129 +0x1b7
main.main()
        ./main.go:8 +0x17
```

**Expected behaviour**
* No panic
* Clear error message is displayed to indicate that the session is not set

--- a/pkg/cmd/factory/c8yclient.go
+++ b/pkg/cmd/factory/c8yclient.go
@@ -29,7 +29,7 @@ var EnvCumulocityHostNames = []string{
 func GetHostFromEnvironment() string {
 	var value = ""
 	for _, name := range EnvCumulocityHostNames {
-		value = os.Getenv(name)
+		value = strings.TrimSpace(os.Getenv(name))
 		if value != "" {
 			break
 		}
@@ -146,6 +146,14 @@ func CreateCumulocityClient(f *cmdutil.Factory, sessionFile, username, password 
 
 		if domain := cfg.GetDomain(); domain != "" {
 			client.SetDomain(domain)
+		}
+
+		if client == nil {
+			return nil, fmt.Errorf("failed to create client")
+		}
+
+		if client.BaseURL == nil {
+			return nil, fmt.Errorf("invalid client url")
 		}
 
 		client.SetRequestOptions(c8y.DefaultRequestOptions{

--- a/pkg/config/cliConfiguration.go
+++ b/pkg/config/cliConfiguration.go
@@ -562,7 +562,7 @@ func (c *Config) GetTenant() string {
 
 // GetHost returns the Cumulocity host URL
 func (c *Config) GetHost() string {
-	return c.viper.GetString("host")
+	return strings.TrimSpace(c.viper.GetString("host"))
 }
 
 // GetDomain gets the custom Cumulocity domain for cases where it differs from the Host

--- a/tests/manual/sessions/sessions_set.yaml
+++ b/tests/manual/sessions/sessions_set.yaml
@@ -183,3 +183,25 @@ tests:
     stdout:
       exactly: |
         {"defaults":{"pagesize":110},"includeall":{"pagesize":202}}
+
+  It does not panic when the url is only contains whitespace:
+    config:
+      env:
+        C8Y_HOST: ' '
+        C8Y_URL: ''
+        C8Y_BASEURL: ''
+        C8Y_SESSION: ''
+        C8Y_USER: ''
+        C8Y_USERNAME: ''
+        C8Y_TENANT: ''
+        C8Y_TOKEN: ''
+        C8Y_PASSWORD: ''
+      inherit-env: false
+    command: |
+      c8y devices list
+    exit-code: 102
+    stderr:
+      not-contains:
+        - panic
+      contains:
+        - A c8y session has not been loaded. Please create or activate a session and try again

